### PR TITLE
Implemented hard sigmoid

### PR DESCRIPTION
**Pull request by PENGUINLIONG**
Not yet tested.

### CODE REVIEW

**Comment by tencent-adm** [comment]
[CLA assistant check](https://cla.opensource.tencent.com/Tencent/ncnn?pullRequest=1046) <br/>Thank you for your submission, we really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla.opensource.tencent.com/Tencent/ncnn?pullRequest=1046) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla.opensource.tencent.com/check/Tencent/ncnn?pullRequest=1046) it.</sub>

File: src/CMakeLists.txt [hunk 1]
```diff
@@ -229,6 +229,7 @@ ncnn_add_layer(Reshape)
 ncnn_add_layer(ROIPooling)
 ncnn_add_layer(Scale)
 ncnn_add_layer(Sigmoid)
+ncnn_add_layer(HardSigmoid)
```
Comment (nihui) [inline src/CMakeLists.txt]: new layer should be appended at the end of this list, for layer type index compability

File: src/layer/hardsigmoid.cpp [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+    int size = w * h;
+
+    #pragma omp parallel for num_threads(opt.num_threads)
+    for (int q=0; q<channels; q++)
+    {
+        float* ptr = bottom_top_blob.channel(q);
+
+        for (int i=0; i<size; i++)
+        {
+            if (ptr[i] < -2.5f)
```
Comment (nihui) [inline src/layer/hardsigmoid.cpp]: y = max(0, min(1, alpha * x + beta))
the alpha and beta value should be parameterized, and loaded from param file
https://github.com/onnx/onnx/blob/master/docs/Operators.md#HardSigmoid

File: src/layer/arm/hardsigmoid_arm.cpp [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+            float32x4_t _ans = vdupq_n_f32(0.5f);
+            _ans = vmlaq_n_f32(_ans, _p, 0.2f);
+            _ans = vmaxq_f32(_ans, _zero);
+            _ans = vminq_f32(_ans, _one);
+            vst1q_f32(ptr, _ans);
+
+            ptr += 4;
+        }
+#endif // __ARM_NEON
+        for (int i=0; i<remain; i++)
```
Comment (nihui) [inline src/layer/arm/hardsigmoid_arm.cpp]: tail loop index is wrong
you can refer https://github.com/Tencent/ncnn/blob/master/src/layer/arm/sigmoid_arm.cpp to learn how to handle the main vectorized loop and remaining data properly

Comment (PENGUINLIONG) [reply src/layer/arm/hardsigmoid_arm.cpp]: Not certain about this. Is the currect implementation efficient?

**Pull request merged**